web-common/src/features/canvas/components/charts/circular-charts/CircularChart.ts

```diff
@@ -0,0 +1,179 @@
+import { derived, get, type Readable } from "svelte/store";
+import type {
+  CanvasEntity,
+  ComponentPath,
+} from "../../../stores/canvas-entity";
+import { BaseChart, type BaseChartConfig } from "../BaseChart";
+import type { ChartDataQuery } from "../types";
+
+type CircularChartEncoding = {
+  measure?: FieldConfig;
+  color?: FieldConfig;
+  innerRadius?: number;
+};
+export type CircularChartSpec = BaseChartConfig & CircularChartEncoding;
+
+export class CircularChartComponent extends BaseChart<CircularChartSpec> {
+  constructor(resource: V1Resource, parent: CanvasEntity, path: ComponentPath) {
+    super(resource, parent, path);
+  }
+
+  protected getChartSpecificOptions(): Record<string, ComponentInputParam> {
+    return {
+      color: {
+        type: "positional",
+        label: "Color",
+        meta: {
+          chartFieldInput: {
+            type: "dimension",
+            nullSelector: true,
+            limitSelector: true,
+            hideTimeDimension: true,
+          },
+        },
+      },
+      measure: {
+        type: "positional",
+        label: "Measure",
+        meta: {
+          chartFieldInput: {
+            type: "measure",
+          },
+        },
+      },
+      innerRadius: {
+        type: "number",
+        label: "Inner Radius",
+      },
+    };
+  }
+
+  createChartDataQuery(
+    ctx: CanvasStore,
+    timeAndFilterStore: Readable<TimeAndFilterStore>,
+  ): ChartDataQuery {
+    const config = get(this.specStore);
+
+    let measures: V1MetricsViewAggregationMeasure[] = [];
+    let dimensions: V1MetricsViewAggregationDimension[] = [];
+
+    if (config.measure?.field) {
+      measures = [{ name: config.measure.field }];
+    }
+
+    let limit: number;
+    let showNull = false;
+    if (config.color?.field) {
+      limit = config.color.limit ?? 20;
+      dimensions = [{ name: config.color.field }];
+      showNull = !!config.color.showNull;
+    }
+
+    return derived(
+      [ctx.runtime, timeAndFilterStore],
+      ([runtime, $timeAndFilterStore], set) => {
+        const { timeRange, where } = $timeAndFilterStore;
+        const enabled = !!timeRange?.start && !!timeRange?.end;
+
+        let mergedWhere = where;
+        if (!showNull && config.color?.field) {
+          const excludeNullFilter = createInExpression(
+            config.color?.field,
+            [null],
+            true,
+          );
+          mergedWhere = mergeFilters(where, excludeNullFilter);
+        }
+
+        const dataQuery = createQueryServiceMetricsViewAggregation(
+          runtime.instanceId,
+          config.metrics_view,
+          {
+            measures,
+            dimensions,
+            where: mergedWhere,
+            timeRange,
+            limit: limit.toString(),
+          },
+          {
+            query: {
+              enabled,
+              placeholderData: keepPreviousData,
+            },
+          },
+          ctx.queryClient,
+        );
```

Comment:
IMO it's quite hard to follow the multiple tiers of derived stores, and they can lead to performance issues. In this case, every change to the `timeAndFilterStore` will re-create a new `dataQuery` observer. These query observers need to be stable for `keepPreviousData` to take effect.

Here's the TanStack Query v5 pattern that I think we should follow:
```typescript
const xQueryOptionsStore = derived(...)
const xQuery = createQuery(xQueryOptionsStore)
return xQuery;
```

In cases where we need to make multiple queries:
```typescript
const xQueryOptionsStore = derived(...)
const yQueryOptionsStore = derived(...)
const zQuery = createQueries({ queries: [xQueryOptionsStore, yQueryOptionsStore], combine: ([xResponse, yResponse] => z))
return zQuery;
```